### DISCUSSION TITLE
how to have markdown-mode configuration specific for eglot/eldoc?

### DISCUSSION THREAD

**Discussion by ville-h** [joaotavora/eglot#1165] [Q&A]
How to have markdown-mode configuration specific for eglot/eldoc information in the minibuffer(?) area?

Honestly it's not entirely clear to me what gadget is responsible for the information display when you put the cursor on a symbol such as a function. At any rate it seems to be shown using markdown-mode. Possibly using [this](https://github.com/joaotavora/eglot/blob/master/eglot.el#L1519)?

The markdown handed out by the lsp I use, clangd for c++, uses some markdown features I don't care to see when viewing the eglot/eldoc information. I would like to see them when I am editing a markdown document. For example in the eglot/eldoc use-case there is no need to make linebreaks caused by two consecutive spaces at the end of the line to be visible in any special way. Which would be possible to achieve by changing the face attributes of markdown-line-break-face. This isn't the only difference I'd like to setup, it's just an example.

**Comment by doolio** [comment]
Duplicate of #1039?

**Reply by joaotavora** [reply]
Yes, and i think there's yet another one around here. Let's make sure they are handled in the same Emacs tracker bug discussion

**Comment by ville-h** [comment]
I am not sure if it's strictly a duplicate. From what I gathered from reading the referenced discussion is that it was about disabling markdown-mode. This is not about disabling markdown-mode. It's about having different settings for different use-cases of markdown-mode: 0) viewing eglot/eldoc information 1) editing a markdown document

One way that comes to mind is that when eglot/eldoc stylizes the buffer with markdown-mode there's some bit of information that could be detected in markdown-mode's hook and thus different settings applied?

**Reply by doolio** [reply]
Well, the other discussion is a suggestion to fall back to plaintext when `markdown-mode` is unable to render certain HTML elements received from some LSP servers. I think your suggestion is to update Eglot to disable the visibility of these elements if they cannot be rendered by `markdown-mode` which is just an alternative solution to the same underlying issue. And as João suggested it is best to combine these suggestions under the same bug discussion on the Emacs bug tracker mailing list.

**Reply by ville-h** [reply]
The elements are rendered "as intended". For some definition of "as intended". I think it's important that it is recognized that the way you might wish markdown to be rendered depends on your use-case at the time:

0) viewing a markdown document
1) authoring/previewing a markdown document
2) something else?

Showing symbol information in the minibuffer would be use-case 0). Anyway I don't suggest eglot/eldoc strip these out, as someone else might want to see them. For my particular use-case setting some specific face-attributes would be enough, but someone else might want to do something more extensive. Maybe be allowed to run their own modification function?

**Comment by ville-h** [comment]
Also if there was the chance to run a user-defined filter function which had the ability to modify the buffer to be displayed that would let the user to determine what and how the information is displayed. One could argue it's the responsibility of the lsp to be configurable in a way to allow the user to select what it returns in the first place. And perhaps that is in some way the ideal situation.

In practice some lsps might evolve those capabilities, but in the meanwhile a practical choice could be to do it on the editor side right before displaying. At least I don't switch the specific lsp for a language very often so it seems like it'd be practical for users to write their filtering functions specific to their environments and get fair bit of mileage out of them.
